### PR TITLE
RFC-460 implemented

### PR DESCRIPTION
**Pull request by juancarlospaco**
- https://github.com/nim-lang/RFCs/issues/460

### CODE REVIEW

File: lib/system/seqs_v2.nim [hunk 1]
```diff
@@ -129,3 +129,19 @@ proc setLen[T](s: var seq[T], newlen: Natural) =
...
+  ## Returns the current capacity of the seq.
+  # See https://github.com/nim-lang/RFCs/issues/460
+  runnableExamples:
+    var lst = newSeqOfCap[string](cap = 42)
+    lst.add "Nim"
+    assert lst.capacity == 42
+
+  {.cast(noSideEffect).}:
+    let sek = unsafeAddr self
+    result = capacityImpl(cast[ptr NimSeqV2](sek)[])
```
Comment (ghost) [inline lib/system/seqs_v2.nim:147]: Is there a reason you can't just do
```nim
result = capacityImpl(cast[NimSeqV2](self))
```
without `sek`?

Comment (juancarlospaco) [reply lib/system/seqs_v2.nim:147]: That does not compile, also I am following the "style" of the rest of the file.

**Comment by Araq** [comment]
Pretty nice but this seems to cover only the new runtime. And there is no `setCapacity` either.

**Comment by juancarlospaco** [comment]
> Pretty nice but this seems to cover only the new runtime. And there is no `setCapacity` either.

@Araq
My idea was to start with the `capacity` only, to keep PR small,
only for new runtime if we are defaulting to ORC soon (?),
giving new features to the new runtime makes more people use it and move over ORC,
because `setLen` also sets the capacity whatsoever.

**Comment by Araq** [comment]
Well supporting only ORC is easier, fair enough. But `setCapacity` is much more important than `capacity` (what would I do with this information if I cannot set the value?)

**Comment by arnetheduck** [comment]
> Well supporting only ORC is easier, fair enough. 

this would prevent writing code that works with both gc:s, for the transitionary period where both are available - this is critical for libraries - in larger projects, it's often not possible to upgrade everything at once and this would make things harder in an unnecessary way.

**Comment by Araq** [comment]
> this would prevent writing code that works with both gc

Sure but I could add the missing pieces afterwards.

**Comment by juancarlospaco** [comment]
> `setCapacity` is much more important than `capacity`

@Araq
I am not sure what to implement for `setCapacity`, I set capacity in the proc and leave the rest to the user ?,
or should it just call the existing `grow()` and `shrink()` procs ?.

I was thinking on just getting read-only capacity, because `setLen` already changes capacity anyways.

Can you provide code or pseudocode or suggestion or some guidance...

File: changelog.md [hunk 2]
```diff
@@ -51,6 +51,12 @@
   making limiting it to just the first char (`last = 0`) valid.
 - `random.rand` now works with `Ordinal`s.
 
+- Added `IsoWeekRange`, a range type to represent the number of weeks in an ISO week-based year.
+- Added `IsoYear`, a distinct int type to prevent bugs from confusing the week-based year and the regular year.
+- Added `initDateTime` in `times` to create a datetime from a weekday, and ISO 8601 week number and week-based year.
+- Added `getIsoWeekAndYear` in `times` to get an ISO week number along with the corresponding ISO week-based year from a datetime.
+- Added `getIsoWeeksInYear` in `times` to return the number of weeks in an ISO week-based year.
```
Comment (Araq) [inline changelog.md]: What?

Comment (juancarlospaco) [reply changelog.md]: Fixed.

**Comment by juancarlospaco** [comment]
@Araq Fixed now, changelog just get conflicts sometimes. 🙂

**Comment by Araq** [comment]
Still needs `setCapacity`.

**Comment by juancarlospaco** [comment]
@Araq 
https://github.com/nim-lang/Nim/pull/19771#issuecomment-1126364905

Again, `setCapacity` is just `setLen`, or some `macro`.

It is not fair with the RFC process, that explicitly says:
" Allow read-only access to cap of seq and string. "

If you have something beyond in your mind, feel free to suggest the code,
because I can not read minds just yet... 🙂

**Comment by Araq** [comment]
> If you have something beyond in your mind, feel free to suggest the code,
because I can not read minds just yet... 🙂

Fair enough, will add the code. It's not the same as `setLen`.

**Comment by juancarlospaco** [comment]
Awesome, I agree, sorry I cant really implement an idea that I dont know about.
For me, I am using it for a macro with `setLen` and `newStringOfCap`. :+1:

**Pull request merged**